Formula/l/llvm.rb

```diff
@@ -412,6 +444,22 @@ def install
       system "/usr/libexec/PlistBuddy", "-c", "Add:CompatibilityVersion integer 2", "Info.plist"
       xctoolchain.install "Info.plist"
       (xctoolchain/"usr").install_symlink [bin, include, lib, libexec, share]
+
+      # Install a major-versioned symlink that can be used across minor/patch version upgrades.
+      xctoolchain.parent.install_symlink xctoolchain.basename.to_s => "LLVM#{soversion}.xctoolchain"
+
+      # Write config files for each macOS major version so that this works across OS upgrades.
+      # TODO: replace this with a call to `MacOSVersion.kernel_major_version` once this is in a release tag:
+      #   https://github.com/Homebrew/brew/pull/18674
+      {
+        11 => 20,
+        12 => 21,
+        13 => 22,
+        14 => 23,
+        15 => 24,
+      }.each do |macos_version, kernel_version|
+        write_config_files(macos_version, kernel_version, Hardware::CPU.arch)
+      end
```

Comment:
Do we already know this formula doesn't work on 10.14 and 10.15? Also would be good to iterate over https://github.com/Homebrew/brew/blob/master/Library/Homebrew/macos_version.rb#L21-L32 automatically.

Reply:
> Do we already know this formula doesn't work on 10.14 and 10.15?

No, but 10.14 doesn't need a config file IIRC.

> Also would be good to iterate over [Homebrew/brew@`master`/Library/Homebrew/macos_version.rb#L21-L32](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/macos_version.rb?rgh-link-date=2024-10-31T08%3A23%3A11Z#L21-L32) automatically.

Yes, but that needs Homebrew/brew@42dd0acdc9ea6f9149b53fe1028e67bb68f653bc which isn't in a release tag yet.

Reply:
> No, but 10.14 doesn't need a config file IIRC.

10.14 is the first version without `/usr/include` (unless you install the header pkg but we do not support/require that). https://github.com/Homebrew/brew/blob/f18e1ea86ab32a8d6119f2d1ae0c6dfcdc77395c/Library/Homebrew/os/mac/xcode.rb#L284